### ISSUE TITLE
LT06 rule not detecting violations in large T-SQL files

### ISSUE THREAD

**Issue by Fank** [quarylabs/sqruff#1772]
# LT06 rule not detecting violations in large T-SQL files

## Description
The LT06 rule (Function name not immediately followed by parenthesis) fails to detect violations in large/complex T-SQL files, even though the same violations are detected correctly in smaller test files.

## Environment
- sqruff version: 0.28.0
- Dialect: tsql

## Steps to Reproduce

1. Create a large T-SQL file (1100+ lines) with functions that have spaces before parentheses, like:
   ```sql
   SELECT COALESCE (value, 0) FROM table;
   ```

2. Run sqruff with LT06 rule:
   ```bash
   sqruff lint file.sql --dialect tsql --rules LT06
   ```

3. No violations are reported.

4. Extract the same problematic code into a smaller file and run the same command - violations ARE detected.

## Expected Behavior
LT06 should detect all instances of functions with spaces before parentheses, regardless of file size or complexity.

## Actual Behavior
LT06 fails to detect violations in large/complex files, while the same code in smaller files triggers the rule correctly.

### Working (detects violation):
```sql
-- test_small.sql
SELECT COALESCE (value, 0) FROM table;
```

### Not working in large file context:
When the same `COALESCE (` pattern appears in a 1100-line file with multiple UNION ALL statements and complex nested queries, the violation is not detected.

## Additional Issue: WITHIN GROUP

The `WITHIN GROUP` clause is not detected by LT06:
```sql
-- This should trigger LT06 but doesn't:
SELECT STRING_AGG(value, ',') WITHIN GROUP (ORDER BY value) FROM table;
```

## Root Cause Analysis
1. Large/complex files may have parsing limitations that prevent LT06 from analyzing all code sections
2. `WITHIN GROUP` may not be recognized as a function-like construct that should follow the same spacing rules

## Impact
This bug allows inconsistent function formatting to pass linting in production SQL files, defeating the purpose of the LT06 rule for maintaining consistent code style.

**Comment by benfdking** [comment]
Yep just looking at it in the playground. It's not parsing the DECLARE statement https://playground.quary.dev/?secondary=Cst#eNqNkmFPwjAQhv/KZV+EBBIB4weMiWNMXDI33WqMccaU7QaLoxttBxrjf/eAEYFE8WN71+d90rtPQ81zo2+02xBgKYukinVWCMB3PitzhLSQ4LLTcxhXE8gEsHZ470aC2h2lKuxvikoXpYIENdJrMYFFVuR8xVHAU40ShrblmoFNjVzjDIVWkVhT2DRTYPmma4eWDctMT0GVPEZwwpqHCYw/1jGRCG3XtthPf2PB8wo7LTg5acJ14N+C5uMcOxc13dwNb+26Lgv5RqaR2JpdxRVVZigVMHPgEtuqL5wEHI81jwnTteezf0t3D6W7W+m/Eniuir2Y3/C9Q3xvi3902I3jwSjwH+6O0aGhsOSSZkZlGjd9Qh0XssDxRq/maLTJo7gW5e3BG34wtAMYPMG6ZdfmjGyMlqFQr/ZF0QI+q7ms0vQlEknGc0qHS9C0m5GQVY6KTnEhkV59fQOo1tz7
